### PR TITLE
Update .travis.yml

### PR DESCRIPTION
**Pull request by mforets**
im using this setup in new projects, i find it useful if JuliaReachDevDocs has it updated too

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,25 +1,33 @@
...
-jobs:
-  include:
-    - stage: "Documentation"
-      julia: 1.0
-      os: linux
-      script:
-        - julia --project=docs/ -e 'using Pkg; Pkg.develop(PackageSpec(path=pwd()));
-                                               Pkg.instantiate();'
-        - julia --project=docs/ docs/make.jl
-      after_success: skip
```
Comment (schillic) [inline .travis.yml:25]: Why remove the building of the documentation? This does not make sense to me.

File: .travis.yml [hunk 2]
```diff
@@ -1,25 +1,33 @@
...
+# https://github.com/travis-ci/travis-ci/issues/4942 workaround
+git:
+  depth: 99999999
```
Comment (schillic) [inline .travis.yml]: The issue got closed, so apparently this is not needed anymore?

File: .travis.yml [hunk 3]
```diff
@@ -1,25 +1,33 @@
...
+
+after_success:
+  - julia --color=yes --project=test/coverage/ -e 'using Pkg; Pkg.instantiate()'
+  - julia --color=yes --project=test/coverage/ test/coverage/coverage.jl
```
Comment (schillic) [inline .travis.yml:29]: Are you sure that this works? Because in other repositories we use just `codecov: true`.

File: .travis.yml [hunk 4]
```diff
@@ -1,25 +1,33 @@
...
 julia:
   - 1.0
+  - 1.2
+  - 1.3
+  - nightly
 
 matrix:
   fast_finish: true
+  allow_failures:
+    - julia: nightly  # development versions
```
Comment (schillic) [inline .travis.yml]:
```suggestion
    - julia: nightly
```

File: .travis.yml [hunk 5]
```diff
@@ -1,25 +1,33 @@
-# Documentation: http://docs.travis-ci.com/user/languages/julia/
+## Documentation: http://docs.travis-ci.com/user/languages/julia/
```
Comment (schillic) [inline .travis.yml]:
```suggestion
# Documentation: http://docs.travis-ci.com/user/languages/julia/
```

File: .travis.yml (hunk 1 above)
Comment (mforets) [reply .travis.yml:25]: This is for building the docs using Github actions (https://juliadocs.github.io/Documenter.jl/stable/man/hosting/#GitHub-Actions-1). I know that we are not using that it in other JuliaReach repos yet.

File: .travis.yml (hunk 2 above)
Comment (mforets) [reply .travis.yml]: ok, yes i think so ... :+1:

File: .travis.yml (hunk 3 above)
Comment (mforets) [reply .travis.yml:29]: i copied this from somewhere else and didnt check what we use in other repositories.

File: .travis.yml (hunk 1 above)
Comment (schillic) [reply .travis.yml:25]: Then I suggest to first try it out.

Comment (mforets) [reply .travis.yml:25]: i'll use this in BernsteinExpansions.jl

Comment (mforets) [reply .travis.yml:25]: i mean this is used in many repositories now, we don't need to wait.. ?

Comment (schillic) [reply .travis.yml:25]: I do not see the problem in trying it out first.

Comment (mforets) [reply .travis.yml:25]: ok, i'll update here when i have used this setup in a project that i manage.

Comment (mforets) [reply .travis.yml:25]: the BernsteinExpansions.jl docs are now built using Github actions workflow (https://juliareach.github.io/BernsteinExpansions.jl/latest/)

Comment (mforets) [reply .travis.yml:25]: should we merge this PR?

File: .travis.yml (hunk 3 above)
Comment (mforets) [reply .travis.yml:29]: i forgot to double check this..

File: .travis.yml (hunk 1 above)
Comment (mforets) [reply .travis.yml:25]: Merging.... i've used this setup now both in BernsteinExpansions and in RangeEnclosures, with docs built ok and codecov true, but note that i added `codecov: true`.

Do open an issue if you have further simplifications / considerations! So i update the other repos too.

**Pull request merged**